src/routes/transactions/mappers/transfers/transfer-imitation.mapper.ts

```diff
@@ -198,10 +196,12 @@ export class TransferImitationMapper {
     // Is imitation if value is lower than the specified threshold
     const value = this.formatValue(txInfo.transferInfo);
     const prevValue = this.formatValue(prevTxInfo.transferInfo);
-    return (
-      // Imitations generally follow high transfer values
-      prevValue >= this.echoLimit && value <= this.echoLimit
-    );
+    const isEchoValue = prevValue >= this.echoLimit && value <= this.echoLimit;
+    if (!isEchoValue) {
+      return false;
+    }
+
+    return this.isImitationAddress(txInfo, prevTxInfo);
```

Comment:
Whilst this check occurs in both methods, I thought it clearer to include it in both.